org.thecrossingchurch.CustomJobs/Jobs/HubspotIntegrationPatching.cs

```diff
@@ -0,0 +1,442 @@
+            key = Encryption.DecryptString(attrValue);
+
+            var current_id = 0;
+
+            PersonService personService = new PersonService(new RockContext());
+
+            //Get Hubspot Properties in Rock Information Group
+            //This will allow us to add properties temporarily to the sync and then not continue to have them forever
+            var propClient = new RestClient("https://api.hubapi.com/crm/v3/properties/contacts?properties=name,label,createdUserId,groupName,options,fieldType");
+            propClient.Timeout = -1;
+            var propRequest = new RestRequest(Method.GET);
+            propRequest.AddHeader("Authorization", $"Bearer {key}");
+            IRestResponse propResponse = propClient.Execute(propRequest);
+            WriteToLog( "Is propResponse Null? : " + propResponse.IsNull().ToString() );
+            var props = new List<HubspotProperty>();
+            var tmbtProps = new List<HubspotProperty>();
+            var propsQry = JsonConvert.DeserializeObject<HSPropertyQueryResult>(propResponse.Content);
+            props = propsQry.results;
+            WriteToLog( "Is propsQry Null? : " + propsQry.IsNull().ToString() );
+            WriteToLog( "Is propsQry.results Null? : " + propsQry.results.IsNull().ToString() );
+            WriteToLog( "Is props Null? : " + props.IsNull().ToString() );
+            
+
+            //Filter to props in Rock Information Group (and Contact information group) 
+            props = props.Where(p => p.groupName == "rock_information" || p.groupName == "contactinformation").ToList();
+            //props = props.Where( p => p.groupName == "Contact information" ).ToList(); // - 6/26 sdlp - To update names, address, and phone
+            //tmbtProps = propsQry.results.Where(p => p.groupName == "rock_tmbt_information").ToList(); // ---------- Commented 6/25 sdlp
+            //Business Unit hs_all_assigned_business_unit_ids
+            //Save a list of the ones that are Rock attributes
+            //var attrs = props.Where(p => p.label.Contains("Rock Attribute ")).ToList(); // commented sdlp 7/15/24
+            RockContext _context = new RockContext();
+            //List<string> attrKeys = attrs.Select(hs => hs.label.Replace("Rock Attribute ", "")).ToList(); // commented sdlp 7/15/24
+            //var rockAttributes = new AttributeService(_context).Queryable().Where(a => a.EntityTypeId == 15 && attrKeys.Contains(a.Key)); //commented sdlp 7/15/24
+
+            Guid transactionTypeGuid = dataMap.GetString("ContributionTransactionType").AsGuid();
+            var transactionTypeDefinedValue = new DefinedValueService(_context).Get(transactionTypeGuid);
+            int transactionTypeValueId = transactionTypeDefinedValue.Id;
+
+            //Get List of all contacts from Hubspot
+            contacts = new List<HSContactResult>();
+            request_count = 0;
+
+            // TODO get all properties...
```

Comment:
Is this To Do comment resolved? If so, can we remove it? If not, what is left to do and does it need to be done in the initial release?